### PR TITLE
chore(sdk): `SealedHeader` generic over header

### PR DESCRIPTION
**Pull request by emhane**
Makes `SealedHeader` type definition generic over header type, which is a necessary first step to use `SealedHeader` in `Block` trait methods

### CODE REVIEW

File: crates/primitives-traits/src/header/sealed.rs [hunk 1]
```diff
@@ -7,29 +7,53 @@ use alloy_rlp::{Decodable, Encodable};
...
-use reth_codecs::add_arbitrary_tests;
+use reth_codecs::{add_arbitrary_tests, Compact};
 use serde::{Deserialize, Serialize};
 
 /// A [`Header`] that is sealed at a precalculated hash, use [`SealedHeader::unseal()`] if you want
 /// to modify header.
 #[derive(Debug, Clone, PartialEq, Eq, Hash, AsRef, Deref, Serialize, Deserialize)]
 #[add_arbitrary_tests(rlp)]
-pub struct SealedHeader {
+pub struct SealedHeader<H = Header> {
```
Comment (mattsse) [inline crates/primitives-traits/src/header/sealed.rs:17]: I think that's reasonable, although for header op and eth use the same type

File: crates/primitives-traits/src/header/sealed.rs [hunk 2]
```diff
@@ -7,29 +7,53 @@ use alloy_rlp::{Decodable, Encodable};
...
+
+    fn from_compact(mut buf: &[u8], _: usize) -> (Self, &[u8]) {
+        let (hash, new_buf) = BlockHash::from_compact(buf, buf.len());
+        buf = new_buf;
+        let (header, new_buf) = H::from_compact(buf, buf.len());
+        buf = new_buf;
+        let sealed_header = Self { hash, header };
+        (sealed_header, buf)
+    }
+}
```
Comment (mattsse) [inline crates/primitives-traits/src/header/sealed.rs:46]: do we even still need this @joshieDo ?

**Review by mattsse — COMMENTED** [review]
q for @joshieDo otherwise lgtm

File: crates/primitives-traits/src/header/sealed.rs (hunk 1 above)
Comment (emhane) [reply crates/primitives-traits/src/header/sealed.rs:18]: we shouldn't only limit this to op

Comment (mattsse) [reply crates/primitives-traits/src/header/sealed.rs:17]: makes sense

**Review by mattsse — APPROVED** [review]

**Pull request merged**